### DISCUSSION TITLE
Vscode "In included file: unknown type name 'concept'"

### DISCUSSION THREAD

**Discussion by meftunca** [jfalcou/eve#1532] [Done & Done!]
`Unknown type name 'concept'clang(unknown_typename)`

Hello, I am writing code with Vs Code. I'm getting this warning about the project. Does anyone know how I can solve this?

**Comment by DenisYaroshevskiy** [comment]
Haven't seen this, I can see

https://github.com/dense-analysis/ale/issues/3443

What do you think?

**Reply by jfalcou** [reply]
> `Unknown type name 'concept'clang(unknown_typename)`
> 
> Hello, I am writing code with Vs Code. I'm getting this warning about the project. Does anyone know how I can solve this?

You need to edit your cpp-settings.json  vsvode configuration to use c++20 instead of c++14 by default.

**Reply by meftunca** [reply]
My c_cpp_properties.json file is :

```json
{
  "configurations": [
    {
      "name": "initial",
      "includePath": [
        "${workspaceFolder}/src",
        "${workspaceFolder}/eve/include",
        "${workspaceFolder}/vendor/**"
      ],
      "compileCommands": "${workspaceFolder}/compile_commands.json",
      "macFrameworkPath": [
        "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks"
      ],
      "compilerPath": "/usr/bin/clang",
      "cppStandard": "c++20",
      "cStandard": "c17",
      "compilerArgs": [],
      "mergeConfigurations": true,
      "browse": {
        "limitSymbolsToIncludedHeaders": false
      }
    }
  ],
  "version": 4
}
```

**Comment by DenisYaroshevskiy** [comment]
Oh no - it's some non vscode thingy

**Comment by DenisYaroshevskiy** [comment]
Do you build some things with eve successfully? Like is at least your compiler/cmake setup working?

I often export compile commands from cmake (I don't remember on the top of my head how) - vscode then picks them up and works with them. That could be a solution

**Reply by meftunca** [reply]
Yes, I'm outputting with cmake and it works with eve in my project. But clang bothers me with such warnings.

I would also like to say that you are a very helpful person. Thank you for that too. You answer my questions promptly with great patience.

**Reply by DenisYaroshevskiy** [reply]
I appreciate you saying that. We are very excited about people trying out our library and are trying to help them succeed.

Unfortunatly at the moment I started to have the same issue with VSCode. I don't know how to solve it.

**Reply by meftunca** [reply]
I solved the problem. I found that the external clangd extension for vscode applies the wrong cpp standard. In Bazel I was able to resolve intellisense errors by forcing all 'copts' settings to -std=c++20. Of course, don't forget to regenerate the 'compile_commands.json' file.
